util/logging.h

```diff
@@ -11,40 +11,47 @@
 // with macros.
 
 #pragma once
-#include "port/port.h"
 
 // Helper macros that include information about file name and line number
-#define STRINGIFY(x) #x
-#define TOSTRING(x) STRINGIFY(x)
-#define PREPEND_FILE_LINE(FMT) ("[" __FILE__ ":" TOSTRING(__LINE__) "] " FMT)
+#define ROCKS_LOG_STRINGIFY(x) #x
+#define ROCKS_LOG_TOSTRING(x) ROCKS_LOG_STRINGIFY(x)
+#define ROCKS_LOG_PREPEND_FILE_LINE(FMT) ("[%s:" ROCKS_LOG_TOSTRING(__LINE__) "] " FMT)
+
+inline const char* ROCKS_LOG_FILENAME(const char* file)
+{
+  // 15 is the length of "util/logging.h"
+  // If the name of this file changed, please change this number, too.
+  return file + sizeof(__FILE__) - 15;
```

Comment:
How about processing the string a little bit and keep the last two segment of the string? For example, if path is `/xx/dd/rrrrr/db/db_impl.cc`, keeps the `db/db_impl.cc` part?

Reply:
Actually, that is what it looks like now.
There is an example:
```
2018/11/01-14:37:21.801951 6b64 [db\db_impl_open.cc:230] Creating manifest 1
2018/11/01-14:37:21.812912 6b64 [db\version_set.cc:3410] Recovering from manifest file: MANIFEST-000001
```

Reply:
Done.
Thanks for your suggestion.